### PR TITLE
Silas add ranking

### PR DESCRIPTION
**Pull request by SilasMarvin**
This integration brings reranking into postgresml.

```
pgml=# SELECT pgml.rank('mixedbread-ai/mxbai-rerank-large-v1', 'test', array_agg(md5(random()::text)), '{"return_documents": false, "top_k": 10}') FROM generate_series(1, 100);
           rank            
---------------------------
 (58,0.20096051692962646,)
 (91,0.2007983922958374,)
 (84,0.1950932741165161,)
 (83,0.1925133764743805,)
 (7,0.1918289214372635,)
 (15,0.1851881593465805,)
 (67,0.18225009739398956,)
 (94,0.1795625537633896,)
 (40,0.17863182723522186,)
 (97,0.177006796002388,)
(10 rows)
```

### CODE REVIEW

File: pgml-extension/src/bindings/transformers/transformers.py [hunk 1]
```diff
@@ -500,6 +500,33 @@ def transform(task, args, inputs, stream=False):
...
+        for x in model.rank(query, documents, **kwargs)
+    ]
+
+
+def rank(transformer, query, documents, kwargs):
+    kwargs = orjson.loads(kwargs)
+
+    if transformer not in __cache_sentence_transformer_by_name:
+        __cache_sentence_transformer_by_name[transformer] = create_cross_encoder(
+            transformer
```
Comment (kczimm) [inline pgml-extension/src/bindings/transformers/transformers.py:523]: If you pass kwargs through to `create_cross_encoder` we can specify the device, https://www.sbert.net/docs/package_reference/cross_encoder/cross_encoder.html?highlight=crossencoder#sentence_transformers.cross_encoder.CrossEncoder

We should do this for the `SentenceTransformer` constructor, too.

**Review by kczimm — COMMENTED** [review]
We should bump the version (probably to 2.9.0 since it's an API addition)  and add the following migration file `sql/pgml--2.8.5--2.9.0.sql`:

```sql
-- src/api.rs:613
-- pgml::api::rank
CREATE  FUNCTION pgml."rank"(
	"transformer" TEXT, /* &str */
	"query" TEXT, /* &str */
	"documents" TEXT[], /* alloc::vec::Vec<&str> */
	"kwargs" jsonb DEFAULT '{}' /* pgrx::datum::json::JsonB */
) RETURNS TABLE (
	"corpus_id" bigint,  /* i64 */
	"score" double precision,  /* f64 */
	"text" TEXT  /* core::option::Option<alloc::string::String> */
)
IMMUTABLE STRICT PARALLEL SAFE 
LANGUAGE c /* Rust */
AS 'MODULE_PATHNAME', 'rank_wrapper';
```

File: pgml-extension/src/bindings/transformers/transformers.py (hunk 1 above)
Comment (SilasMarvin) [reply pgml-extension/src/bindings/transformers/transformers.py:523]: We will have to create a separate argument for this, or pop specific arguments from kwargs. If just pass kwargs straight through we will get an unexpected keyword argument error.

**Comment by SilasMarvin** [comment]
> We should bump the version (probably to 2.9.0 since it's an API addition) and add the following migration file `sql/pgml--2.8.5--2.9.0.sql`:
> 
> ```sql
> -- src/api.rs:613
> -- pgml::api::rank
> CREATE  FUNCTION pgml."rank"(
> 	"transformer" TEXT, /* &str */
> 	"query" TEXT, /* &str */
> 	"documents" TEXT[], /* alloc::vec::Vec<&str> */
> 	"kwargs" jsonb DEFAULT '{}' /* pgrx::datum::json::JsonB */
> ) RETURNS TABLE (
> 	"corpus_id" bigint,  /* i64 */
> 	"score" double precision,  /* f64 */
> 	"text" TEXT  /* core::option::Option<alloc::string::String> */
> )
> IMMUTABLE STRICT PARALLEL SAFE 
> LANGUAGE c /* Rust */
> AS 'MODULE_PATHNAME', 'rank_wrapper';
> ```

Done! https://github.com/postgresml/postgresml/pull/1498/commits/745b190fbfe6b5a0551b8aeab46644c29caa5db0

**Review by kczimm — APPROVED** [review]

**Pull request merged**